src/pybkgmodel/data.py

```diff
@@ -501,23 +537,236 @@ def load_events(cls, file_name, cuts):
             event_data['mjd'],
-            event_data['delta_t']
+            event_data['delta_t'],
+            None
         )
 
         return event_sample
 
+class DL3EventFile(EventFile):
+    """Reader for DL3 data file compliant with the GADF. For details see
+    https://gamma-astro-data-formats.readthedocs.io/
+
+    Parameters
+    ----------
+    file_name: str
+            Name of the DL3 file to use.
+    """
+    def __init__(self, file_name):
+        super().__init__(file_name)
+
+        self.file_name = file_name
+        self.obs_id = self.get_obs_id(file_name)
+        self.events = self.load_events(file_name)
+
+    @classmethod
+    def is_compatible(cls, file_name):
+        """Function checks whether the file is a fits file according to the file extension.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        bool
+            True if fits file according to file extension
+        """
+
+        ext = Path(file_name).suffixes
+        compatible = ".fits" in ext
+
+        return compatible
+
+    @classmethod
+    def get_obs_id(cls, file_name):
+        """Function reads the observation id from the fits file.
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        int
+            Observation ID number of the fits file
+
+        Raises
+        ------
+        RuntimeError
+            In case Observation ID is not found or Events HDU layer is missing. In this case the file is not complient to GADF.
+        """
+
+        obs_id = None
+
+        with fits.open(file_name, memmap=False) as input_file:
+            try:
+                obs_id = int(input_file["EVENTS"].header['OBS_ID'])
+            except Exception as error:
+                raise RuntimeError(f'Can not find observations ID in {file_name}') from error
+
+        return obs_id
+
+    @classmethod
+    def load_events(cls, file_name):
+        """Function to read the events from the DL3 file and to compute the missing variables needed by pybkgmodel
+
+        Parameters
+        ----------
+        file_name: str
+            Name of the DL3 file to use.
+
+        Returns
+        -------
+        dict:
+            A dictionary with the event properties: arrival time, direction, and energy.
+        """
+
+        data_names_mapping = {
+            'EVENT_ID': 'daq_event_number',
+            'RA': 'event_ra',
+            'DEC': 'event_dec',
+            'GAMMANESS': 'gammaness',
+            'ENERGY': 'event_energy',
+        }
+
+        with fits.open(file_name, memmap=False) as input_file, \
+            erfa_astrom.set(ErfaAstromInterpolator(1 * u.s)):
+            try:
+                evt_hdu = input_file["EVENTS"]
+                evt_head = evt_hdu.header
+                evt_data = pandas.DataFrame(evt_hdu.data)
```

Comment:
use an astropy table to have support for the quantities and metadata stored in the hdu

Reply:
Thanks for that hint, simplified it a lot. Adapted.